src/cert/ssl_cert_provider.cpp

```diff
@@ -312,6 +312,13 @@ mp::SSLCertProvider::KeyCertificatePair make_cert_key_pair(const QDir& cert_dir,
         const std::filesystem::path root_cert_path = MP_PLATFORM.get_root_cert_path();
         if (std::filesystem::exists(root_cert_path) && QFile::exists(priv_key_path) && QFile::exists(cert_path))
         {
+            // In cases where the root certificate exists within the data folder (such as in Snap or Windows
+            // environments), its file permissions is unintentionally overwritten due to recursive permission
+            // settings applied to the entire folder. To ensure the client process has access, we need to explicitly
+            // reset the file permissions on the root certificate.
```

Comment:
Hmm I don't think we need all these comments. They are valuable for the PR, not so much for future readers of the code IMO. I mean, permissions are no longer "unintentionally overwritten" once this is merged, right? How about something simpler, like

```suggestion
            // Unlike other daemon files, the root certificate needs to be accessible by everyone
```